pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -131,6 +143,10 @@ func (b *bucketAccessRequestListener) provisionBucketAccess(ctx context.Context,
 		return err
 	}
 
+	if !util.CheckFinalizer(bucketAccessRequest, util.BARDeleteFinalizer) {
+		bucketAccessRequest.ObjectMeta.Finalizers = append(bucketAccessRequest.ObjectMeta.Finalizers, util.BARDeleteFinalizer)
```

Comment:
Does this append work if metadata.Finalizers is nil (empty)?

Reply:
should be fine as it is empty array.